### ISSUE TITLE
Perpetual Market Making - Unexpected error while processing event 202 during funding payment period

### ISSUE THREAD

**Issue by rxlxrxsx** [hummingbot/hummingbot#5084]
**Describe the bug**
Getting an `Unexpected error while processing event 202` during funding payment period when running a perpetual market making strategy with a position currently open. This also results to funding payment not getting recorded on the sqlite file.

```python
2022-02-03 00:00:30,313 - 949 - hummingbot.connector.derivative.binance_perpetual.binance_perpetual_derivative - ERROR - Unexpected error while processing event 202.
Traceback (most recent call last):
  File "hummingbot/core/pubsub.pyx", line 165, in hummingbot.core.pubsub.PubSub.c_trigger_event
    typed_listener.c_call(arg)
  File "hummingbot/core/event/event_listener.pyx", line 25, in hummingbot.core.event.event_listener.EventListener.c_call
    self(arg)
  File "/home/ubuntu/hummingbot/hummingbot/core/event/event_forwarder.py", line 24, in __call__
    self._to_function(self.current_event_tag, self.current_event_caller, arg)
  File "/home/ubuntu/hummingbot/hummingbot/connector/markets_recorder.py", line 286, in _did_complete_funding_payment
    session: Session = self.session
AttributeError: 'MarketsRecorder' object has no attribute 'session'
```

**Steps To Reproduce**
1. Create and run a perpetual market making strategy
2. Set a tight spread, wait for an order to get filled to open a position
3. Once a position is open wait for the funding payment period

**Screenshots**
<img width="1422" alt="Screen Shot 2022-02-02 at 4 00 45 PM" src="https://user-images.githubusercontent.com/64377055/152269348-2dcb096f-fb29-47f1-ad15-c29665e71209.png">

**Release version**
1.0.0

**Attachments**
Logs/config: [perp_mm_funding.zip](https://github.com/hummingbot/hummingbot/files/7991270/perp_mm_funding.zip)

**Comment by JulesText** [comment]
This issue is also occurring for me on the dydx_perpetual_derivative exchange.

strategy: perpetual_market_making
template_version: 6
release version: dev-1.0.0

[errors.zip](https://github.com/hummingbot/hummingbot/files/8195919/errors.zip)

**Comment by rheza** [comment]
Happening again after removed the session line.

`
2022-04-17 16:00:30,142 - 24244 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1650124800000, "market": "binance_perpetual", "trading_pair": "BTC-USDT", "amount": "-0.00248301", "funding_rate": "0.00010000", "event_name": "FundingPaymentCompletedEvent", "event_source": "binance_perpetual"}
2022-04-17 16:00:30,142 - 24244 - hummingbot.connector.derivative.binance_perpetual.binance_perpetual_derivative - INFO - Funding payment of 0.00029060 received on BTC-USDT market.
2022-04-17 16:00:30,142 - 24244 - hummingbot.connector.derivative.binance_perpetual.binance_perpetual_derivative - ERROR - Unexpected error while processing event 202.
Traceback (most recent call last):
  File "hummingbot/core/pubsub.pyx", line 165, in hummingbot.core.pubsub.PubSub.c_trigger_event
    typed_listener.c_call(arg)
  File "hummingbot/core/event/event_listener.pyx", line 25, in hummingbot.core.event.event_listener.EventListener.c_call
    self(arg)
  File "/home/azureuser/hummingbot/hummingbot/core/event/event_forwarder.py", line 24, in call
    self._to_function(self.current_event_tag, self.current_event_caller, arg)
  File "/home/azureuser/hummingbot/hummingbot/connector/markets_recorder.py", line 284, in _did_complete_funding_payment
    session: Session = self.session
AttributeError: 'MarketsRecorder' object has no attribute 'session'
`

anyone got any fix for this problem ?

**Comment by JulesText** [comment]
Hi there @rheza 

The error log you're showing seems to indicate the session line that was removed in the PR mentioned above is still there in your code. The last line in your log includes the code:
`session: Session = self.session`
However that is the line that causes the error.
If you're working from source and edited the code file to remove the line, can I suggest recompiling your code before running hummingbot again? The command for this is:
`./compile`

**Comment by rapcmia** [comment]
The fix is applied on the Hummingbot version 1.4 release. Thank you
